wremnants/CardTool.py

```diff
@@ -355,30 +379,64 @@ def variationName(self, proc, name):
         else:
             return f"{self.histName}_{proc}_{name}"
 
-    def checkSysts(self, hnom, var_map, proc, thresh=0.25):
+    def getBoostHistByCharge(self, h, q):
+        return h[{"charge" : h.axes["charge"].index(q) if q != "sum" else hist.sum}]
+        
+    def checkSysts(self, hnom3D, var_map, proc, thresh=0.25):
         #if self.check_variations:
         var_names = set([name.replace("Up", "").replace("Down", "") for name in var_map.keys() if name])
         if len(var_names) != len(var_map.keys())/2:
             raise ValueError(f"Invalid syst names for process {proc}! Expected an up/down variation for each syst. "
                 f"Found systs {var_names} and outNames {var_map.keys()}")
         for name in var_names:
-            up = var_map[name+"Up"]
-            down = var_map[name+"Down"]
-            try:
-                up_relsign = np.sign(up.values()-hnom.values())
-            except ValueError as e:
-                logger.error(f"Incompatible shapes between up and down for syst {name}")
-                raise e
-            down_relsign = np.sign(down.values()-hnom.values())
-            vars_sameside = (up_relsign != 0) & (up_relsign == down_relsign)
-            perc_sameside = np.count_nonzero(vars_sameside)/hnom.size 
-            if perc_sameside > thresh:
-                logger.warning(f"{perc_sameside:.0%} bins are one sided for syst {name} and process {proc}!")
+            for chan in self.channels:
+                if not self.keepOtherChargeSyst and self.chargeIdDict[chan]["badId"] is not None and self.chargeIdDict[chan]["badId"] in name:
+                    continue
+                if chan in ["plus", "minus"]:
+                    q = self.chargeIdDict[chan]["val"]
+                    hnom = self.getBoostHistByCharge(hnom3D, q)
+                    up = self.getBoostHistByCharge(var_map[name+"Up"], q)
+                    down = self.getBoostHistByCharge(var_map[name+"Down"], q)
+                else:
+                    hnom = hnom3D
+                    up = var_map[name+"Up"]
+                    down = var_map[name+"Down"]
+                nCellsWithoutOverflows = 1
```

Comment:
nCellsWithoutOverflows = np.product(nom.shape)

Reply:
Actually at this point it would probably be good to put these ~10 lines into their own function (check_syst_bands or something)